### ISSUE TITLE
hashlib.md5 doesn't work when FIPS is enabled

### ISSUE THREAD

**Issue by dantebben** [pytest-dev/pytest-randomly#414]
### Python Version

3.9.9

### pytest Version

6.2.5

### Package Version

3.10.2

### Description

On a system where FIPS is enabled, pytest-randomly fails on line 275 
```python
hasher = hashlib.md5()
```
with the following error
```text
ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS
```

This appears to be common among tools that use `hashlib`.  The following workaround seems to fix the problem, edit `pytest_randomly/__init__.py` line 275 to be:
```python
hasher = hashlib.md5(usedforsecurity=False)
```

I found this idea for a fix here:  https://github.com/s3tools/s3cmd/issues/1005#issuecomment-578241131.  This issue was related to another package using `hashlib.md5`

**Comment by johnthagen** [comment]
Note that `usedforsecurity` was introduced in 3.9: https://docs.python.org/3/library/hashlib.html#hash-algorithms

Some kind of conditional will be needed to support older versions of Python.

https://github.com/pytest-dev/pytest-randomly/blob/fbb03b718e74fb78baebd936f94298ee17e2449c/.github/workflows/main.yml#L16-L21

**Comment by adamchainz** [comment]
Thanks for the report. This has been fixed in #415 and released in version 3.10.3. Enjoy.

**Comment by dantebben** [comment]
awesome!  thanks.
